keyboards/planck/keymaps/adamtabrams/keymap.c

```diff
@@ -208,7 +206,108 @@ const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
 
 };
 
+bool is_left_mod(uint16_t keycode) {
+    switch (keycode) {
+        case ALT__A:
+        case GUI__S:
+        case SFT__D:
+        case CTL__F:
+        case GUI__R:
+        case SFT__S:
+        case CTL__T:
+            return true;
+    }
+    return false;
+}
+
+bool is_right_mod(uint16_t keycode) {
+    switch (keycode) {
+        case ALTSCLN:
+        case GUI__L:
+        case SFT__K:
+        case CTL__J:
+        case ALT__O:
+        case GUI__I:
+        case SFT__E:
+        case CTL__N:
+            return true;
+    }
+    return false;
+}
+
+bool is_left_key(int row) {
+    return row < 4 || row == 7;
+}
+
+bool is_right_key(int row) {
+    return row > 2;
+}
```

Comment:
This part is actually specific to planck rev6 (and rev7, which apparently has the same matrix).  And technically the handling of rows 3 and 7 is wrong (row 3 is used for both far left and far right keys on the bottom row, and row 7 is used by the central 6 keys of the bottom row).